src/dash-table/components/EdgeFactory.tsx

```diff
@@ -0,0 +1,228 @@
+            return target;
+        }
+
+        target = target.clone();
+
+        const hNext = next.getMatrices().horizontal;
+        const hTarget = target.getMatrices().horizontal;
+
+        const iNext = 0;
+        const iTarget = hTarget.rows - 1;
+
+        R.forEach(j =>
+            (
+                hNext.getWeight(iNext, j) > cutoffWeight ||
+                (hNext.getWeight(iNext, j) > hTarget.getWeight(iTarget, j))
+            ) && hTarget.setEdge(iTarget, j, undefined, -Infinity, true),
+            R.range(0, hTarget.columns)
+        );
+
+        return target;
+    }
+
+    private vReconcile(target: EdgesMatrices | undefined, next: EdgesMatrices | undefined, cutoffWeight: number): EdgesMatrices | undefined {
+        if (!target || !next) {
+            return target;
+        }
+
+        target = target.clone();
+
+        const vNext = target.getMatrices().vertical;
+        const vTarget = target.getMatrices().vertical;
+
+        const jNext = 0;
+        const jTarget = vTarget.columns - 1;
+
+        R.forEach(i =>
+            (
+                vNext.getWeight(i, jNext) > cutoffWeight ||
+                (vNext.getWeight(i, jNext) > vTarget.getWeight(i, jTarget))
+            ) && vTarget.setEdge(i, jTarget, undefined, -Infinity, true),
+            R.range(0, vTarget.rows)
+        );
+
+        return target;
+    }
+
+    private get props() {
+        return this.propsFn();
+    }
+
+    constructor(private readonly propsFn: () => ControlledTableProps) {
+
+    }
+
+    public createEdges() {
+        const {
+            columns,
+            filtering,
+            row_deletable,
+            row_selectable,
+            style_cell,
+            style_cell_conditional,
+            style_data,
+            style_data_conditional,
+            style_filter,
+            style_filter_conditional,
+            style_header,
+            style_header_conditional,
+            virtualized
+        } = this.props;
+
+        return this.memoizedCreateEdges(
+            columns,
+            (row_deletable ? 1 : 0) + (row_selectable ? 1 : 0),
+            !!filtering,
+            style_cell,
+            style_cell_conditional,
+            style_data,
+            style_data_conditional,
+            style_filter,
+            style_filter_conditional,
+            style_header,
+            style_header_conditional,
+            virtualized.data,
+            virtualized.offset
+        );
+    }
+
+    private memoizedCreateEdges = memoizeOne((
+        columns: VisibleColumns,
+        operations: number,
+        filtering: boolean,
+        style_cell: Style,
+        style_cell_conditional: Cells,
+        style_data: Style,
+        style_data_conditional: DataCells,
+        style_filter: Style,
+        style_filter_conditional: BasicFilters,
+        style_header: Style,
+        style_header_conditional: Headers,
+        data: Data,
+        offset: IViewportOffset
+    ) => {
```

Comment:
The overall edges calculation is expensive and cached.
The specific edges calculation are cached.
The style calculation for each section are cache.